### PR TITLE
[PRTL-2918] refactor continuous card

### PR DESCRIPTION
**Pull request by samrichca**
## Ticket Number

PRTL-2918

## Environment to be used in testing

- [x] `prod`
- [ ] `dev-oicr`
- [ ] `qa` (which version?)

### CODE REVIEW

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/utils/continuous.js [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  });
+};
+
+const parseContinuousValue = continuousValue =>
+  Number(Number(continuousValue).toFixed(2));
+
+export const parseContinuousKey = keyValue =>
+  keyValue.split('-')
+    .map((val, idx, src) => (src[idx - 1] === '' ? `-${val}` : val))
+    .filter(val => val !== '')
```
[the remark this answers is not in the document]
Comment (samrichca) [reply src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/utils/continuous.js]: good suggestion!

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/utils/continuous.js [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+  Max: 'Maximum',
+  Min: 'Minimum',
+  SD: 'Standard Deviation',
+};
+
+export const makeContinuousSetFilters = (selectedBuckets, fieldName, filters) => {
+  const bucketRanges = selectedBuckets.map(bucket => bucket.rangeValues);
+
+  return addInFilters(filters, {
+    content: bucketRanges.length === 1 && bucketRanges[0].max === -1
```
[the remark this answers is not in the document]
Comment (samrichca) [reply src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/utils/continuous.js]: the weird thing is, `makeCategoricalSetFilters` and `makeContinuousSetFilters` are only called when `selectedBuckets.length === 0`. I haven't been able to trigger them, even when there's no data for the cards.

I'm not quite sure what this does, we can chat about this offline when I'm back in the office

[the remark this answers is not in the document]
Comment (yuanhengzhao) [reply src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/utils/continuous.js]: Sure.

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/utils/categorical.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+export const makeCategoricalSetFilters = (selectedBuckets, fieldName, filters) => {
+  const bucketFilters = []
```
[the remark this answers is not in the document]
Comment (samrichca) [reply src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/utils/categorical.js]: see below

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/utils/continuous.js [hunk 4]
```diff
@@ -79,26 +78,26 @@ const parseContinuousValue = continuousValue =>
 
 export const parseContinuousKey = keyValue =>
   keyValue.split('-')
-    .map((val, idx, src) => (src[idx - 1] === '' ? `-${val}` : val))
-    .filter(val => val !== '')
-    .map(val => parseContinuousValue(val));
+    .map((val, idx, src) => (parseContinuousValue(src[idx - 1] === ''
```
Comment (yuanhengzhao) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/utils/continuous.js]: ` .filter(val => val !== '')` is missing.

Comment (samrichca) [reply src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/utils/continuous.js]: DOH thank you

**Pull request merged**